transports/sftp/src/main/java/org/mule/transport/sftp/SftpClient.java

```diff
@@ -214,6 +220,19 @@ public void login(String user, String identityFile, String passphrase) throws IO
         }
     }
 
+    private void configureProxy(Session session)
+    {
+        if (proxyConfig != null)
+        {
+            ProxyHTTP proxy = new ProxyHTTP(proxyConfig.getHost(), proxyConfig.getPort());
```

Comment:
ProxyHTTP is not the most common use case. User should be able to choose from SocksS5, SocksS4 and Http